### ISSUE TITLE
Refactor the physics engine's hierarchy

### ISSUE THREAD

**Issue by John-WL** [John-WL/EffectiveCollisionsGame#3]
We currently have 2 classes for the engine: a collision detection and an optimized dynamic body actualizer.

Shouldn't we embed these two in somwthing like a "physics engine" class or something of the sort? Should we seperate dynamic bodies and their canevas prints in different classes?

I saw that at some point that there is two different classes for cirecles: "Circle", and "CircleData". I assume one is for the gui print and the other is for the position and radius? That's part of what I'm talking about. 

What do you think?

**Comment by John-WL** [comment]
I re-read my first comment and it doesn't seem clear. Let me rephrase it. There is 2 different questions here. 

I think we should refactor the physic's engine class hierarchy. Should we embed the collision detector and the body actualizer into another class than the one we're currently using? 

Also, I think that we should seperate the data of the bodies from their gui behaviour. 

What do you think?

**Comment by lebel-louisjacob** [comment]
Assuming that by "gui behaviour", you mean visual object movements:

I guess we should separate the code for the gui from the the code for the backend. But maybe there are other things that matter more right now.

For the organization of the collision detection mechanism and the objects actualiser, hasn't this been already implemented?

**Comment by John-WL** [comment]
by gui behaviour, I mean gui prints and access

**Comment by John-WL** [comment]
The super class that embeds both the collisionner and the updater is innexistant

**Comment by lebel-louisjacob** [comment]
> The super class that embeds both the collisionner and the updater is innexistant

well, isn't the class `EffectiveCollisions` exactly having 2 private fields `CollisionHandler` and `ParticleHandler`, and isn't it not only using those in its methods?

**Comment by John-WL** [comment]
yes, you are absolutely right about that. I would simply prefer to be able to reuse the architecture we're building right now in other projects of the sort.

**Comment by John-WL** [comment]
or then, should we rename "effectiveCollisions" to "physicsEngine"?

**Comment by John-WL** [comment]
Est-ce qu'on peut ferner cette issue? Je pense pad que nous allons travailler dessus de si tôt

**Comment by John-WL** [comment]
et elle prend de la place pour rien dans mes issues

**Comment by John-WL** [comment]
J'imagine que je pourrais faire comme si elle était pas là

**Comment by John-WL** [comment]
mais à chaque fois que je la vois ça me rend nostalgique somehow xD

**Comment by John-WL** [comment]
le bon vieux temps où on voulait créer un engine de jeu vidéo dans processing... ah
